### PR TITLE
Eth2Near-relay: split relay on a few crates, many improvements

### PR DESCRIPTION
**Pull request by olga24912**
Separate logger, eth_rpc_client and Eth_client_initialization

* check configuration setup for mainnet: https://github.com/aurora-is-near/rainbow-bridge/issues/767
* config for initialization https://github.com/aurora-is-near/rainbow-bridge/issues/769
* use enum for some args in config https://github.com/aurora-is-near/rainbow-bridge/issues/811
* unwrap -> expect https://github.com/aurora-is-near/rainbow-bridge/issues/816

Not fixed:
* don't panic https://github.com/aurora-is-near/rainbow-bridge/issues/815

PR should be merged after https://github.com/aurora-is-near/rainbow-bridge/pull/818 is merged.

### CODE REVIEW

**Comment by sept-en** [comment]
@olga24912 please fix conflicts with `master` branch

File: eth2near/eth2-contract-init/src/init_contract.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                .execution_payload
+                .block_number,
+        )
+        .expect("Error on fetching finalized execution header");
+
+    let next_sync_committee = light_client_update
+        .sync_committee_update
+        .expect("No sync_committee update in light client update")
+        .next_sync_committee;
+    let prev_light_client_update = beacon_rpc_client.get_light_client_update(period - 1)?;
```
[the remark this answers is not in the document]
Comment (olga24912) [reply eth2near/eth2-contract-init/src/init_contract.rs]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/a2dc229267d2075efababe7b2dc21ce17fa555a4

**Review by karim-en — CHANGES_REQUESTED** [review]
Please merge with master to fix the conflicts

File: eth2near/eth2-contract-init/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    // Timeout for ETH RPC requests in seconds
+    pub eth_requests_timeout_seconds: u64,
+
+    pub validate_updates: bool,
+
+    pub verify_bls_signature: bool,
+
+    pub hashes_gc_threshold: u64,
+
+    pub max_submitted_blocks_by_account: u32,
```
Comment (karim-en) [inline eth2near/eth2-contract-init/src/config.rs:47]:
```
eth_requests_timeout_seconds: 10,
validate_updates: true,
verify_bls_signatures: false,
hashes_gc_threshold: 51_000,
max_submitted_blocks_by_account: 8000,
```
            
I think that it is better to make those configs optional with default values

File: eth2near/contract_wrapper/src/lib.rs [hunk 3]
```diff
@@ -8,4 +8,7 @@ pub mod eth_client_contract_trait;
 pub mod file_eth_client_contract;
 pub mod near_contract_wrapper;
 pub mod sandbox_contract_wrapper;
-pub mod utils;
\ No newline at end of file
+pub mod utils;
+pub mod eth_network_enum;
+pub mod near_network_enum;
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/lib.rs:13]: I think `enum` suffix is redundant. To be removed in future PRs.

File: eth2near/logger/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "logger"
```
Comment (sept-en) [inline eth2near/logger/Cargo.toml:2]:
```suggestion
name = "eth2near-logger"
```

File: eth2near/eth2near-block-relay-rs/Cargo.toml [hunk 5]
```diff
@@ -32,6 +32,7 @@ hex = "*"
 toml = "0.5.9"
 atomic_refcell = "0.1.8"
 bitvec = "*"
+logger = { path = "../logger" }
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/Cargo.toml:35]:
```suggestion
eth2near-logger = { path = "../logger" }
```

File: eth2near/contract_wrapper/src/eth_network_enum.rs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct IncorrectEthNetwork;
+
+impl Display for IncorrectEthNetwork {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Unknown Eth network. Possible Eth network: 'Mainnet', 'Kiln', 'Goerli', 'Ropsten'"
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/eth_network_enum.rs:22]:
```suggestion
            "Unknown Ethereum network. Possible networks: 'Mainnet', 'Kiln', 'Goerli', 'Ropsten'"
```

File: eth2near/contract_wrapper/src/eth_network_enum.rs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+impl Error for IncorrectEthNetwork {}
+
+impl Display for EthNetwork {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            EthNetwork::Mainnet => write!(f, "mainnet"),
+            EthNetwork::Kiln => write!(f, "kiln"),
+            EthNetwork::Goerli => write!(f, "goerli"),
+            EthNetwork::Ropsten => write!(f, "ropsten")
+        }
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/eth_network_enum.rs:36]: No need to duplicate the names of networks as `as_str()` is already implemented.

```suggestion
        write!(f, "{}", self.as_str())
```

File: eth2near/contract_wrapper/src/eth_network_enum.rs [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    type Err = IncorrectEthNetwork;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Mainnet" | "mainnet" => Ok(EthNetwork::Mainnet),
+            "Kiln" | "kiln" => Ok(EthNetwork::Kiln),
+            "Goerli" | "goerli" => Ok(EthNetwork::Goerli),
+            "Ropsten" | "ropsten" => Ok(EthNetwork::Ropsten),
+            _ => Err(IncorrectEthNetwork),
+        }
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/eth_network_enum.rs:61]: It's better to parse the lowercase version, which allows more invariants.

```suggestion
        match s.to_lowercase().as_str() {
            "mainnet" => Ok(EthNetwork::Mainnet),
            "kiln" => Ok(EthNetwork::Kiln),
            "goerli" => Ok(EthNetwork::Goerli),
            "ropsten" => Ok(EthNetwork::Ropsten),
            _ => Err(IncorrectEthNetwork),
        }
```

File: eth2near/contract_wrapper/src/near_network_enum.rs [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+impl Error for IncorrectNearNetwork {}
+
+impl Display for NearNetwork {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            NearNetwork::Mainnet => write!(f, "mainnet"),
+            NearNetwork::Testnet => write!(f, "testnet"),
+        }
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/near_network_enum.rs:32]: No need to duplicate the names of networks as `as_str()` is already implemented.

```suggestion
        write!(f, "{}", self.as_str())
```

File: eth2near/contract_wrapper/src/near_network_enum.rs [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+impl FromStr for NearNetwork {
+    type Err = IncorrectNearNetwork;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Mainnet" | "mainnet" => Ok(NearNetwork::Mainnet),
+            "Testnet" | "testnet" => Ok(NearNetwork::Testnet),
+            _ => Err(IncorrectNearNetwork),
+        }
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/near_network_enum.rs:53]: It's better to parse the lowercase version, which allows more invariants.

```suggestion
        match s.to_lowercase().as_str() {
            "mainnet" => Ok(NearNetwork::Mainnet),
            "testnet" => Ok(NearNetwork::Testnet),
            _ => Err(IncorrectNearNetwork),
        }
```

File: eth2near/contract_wrapper/src/near_network_enum.rs [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct IncorrectNearNetwork;
+
+impl Display for IncorrectNearNetwork {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Unknown near network id. Possible near networks: 'Mainnet', 'Testnet'"
```
Comment (sept-en) [inline eth2near/contract_wrapper/src/near_network_enum.rs:20]:
```suggestion
            "Unknown NEAR network id. Possible networks: 'Mainnet', 'Testnet'"
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 12]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
     }
 
     fn check_urls(&self) {
         // check `beacon_endpoint`
-        Url::parse(&self.beacon_endpoint).unwrap();
+        Url::parse(&self.beacon_endpoint).expect("Error on beacon endpoint URL parsing");
 
         // check `eth1_endpoint`
-        Url::parse(&self.eth1_endpoint).unwrap();
+        Url::parse(&self.eth1_endpoint).expect("Error on eth1 endpoint URL parsing");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:105]:
```suggestion
        Url::parse(&self.eth1_endpoint).expect("Error on ETH1 endpoint URL parsing");
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 13]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
 
         // check `near_endpoint`
-        Url::parse(&self.near_endpoint).unwrap();
+        Url::parse(&self.near_endpoint).expect("Error on near endpoint URL parsing");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:108]:
```suggestion
        Url::parse(&self.near_endpoint).expect("Error on NEAR endpoint URL parsing");
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 14]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
     }
 
     fn check_account_id(&self) {
         let near_rpc_client = NearRPCClient::new(&self.near_endpoint);
 
         // check `signer_account_id`
-        let _signer_account_id: near_sdk::AccountId = self.signer_account_id.parse().unwrap();
+        let _signer_account_id: near_sdk::AccountId = self.signer_account_id.parse().expect("Error on signature account id parsing");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:115]:
```suggestion
        let _signer_account_id: near_sdk::AccountId = self.signer_account_id.parse().expect("Error on signer account ID parsing");
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 15]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
         if !near_rpc_client
             .check_account_exists(&self.signer_account_id)
-            .unwrap()
+            .expect("Error on checking signature account id existence")
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:118]:
```suggestion
            .expect("Error on checking signer account ID existence")
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 16]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
         {
             panic!("Signer account id doesn't exist on NEAR network");
         }
 
         // check `contract_account_id`
-        let _contract_account_id: near_sdk::AccountId = self.contract_account_id.parse().unwrap();
+        let _contract_account_id: near_sdk::AccountId = self.contract_account_id.parse().expect("Error on contract account id parsing");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:124]:
```suggestion
        let _contract_account_id: near_sdk::AccountId = self.contract_account_id.parse().expect("Error on contract account ID parsing");
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 17]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
         if !near_rpc_client
             .check_account_exists(&self.contract_account_id)
-            .unwrap()
+            .expect("Error on checking contract account id existence")
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:127]:
```suggestion
            .expect("Error on checking contract account ID existence")
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 18]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
         {
             panic!("Contract account id doesn't exist on NEAR network");
         }
 
         // check `dao_contract_account_id`
         if let Some(dao_contract_account_id) = self.dao_contract_account_id.clone() {
             let _dao_contract_account_id: near_sdk::AccountId =
-                dao_contract_account_id.parse().unwrap();
+                dao_contract_account_id.parse().expect("Error on dao contract account id parsing");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:135]:
```suggestion
                dao_contract_account_id.parse().expect("Error on DAO contract account ID parsing");
```

File: eth2near/eth2near-block-relay-rs/src/config.rs [hunk 19]
```diff
@@ -83,84 +86,59 @@ pub struct Config {
...
             if !near_rpc_client
                 .check_account_exists(&dao_contract_account_id)
-                .unwrap()
+                .expect("Error on checking DAO account id existence")
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/config.rs:138]:
```suggestion
                .expect("Error on checking DAO account ID existence")
```

File: eth2near/eth2near-block-relay-rs/src/contract_type.rs [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+
+impl Error for IncorrectContractType {}
+
+impl Display for ContractType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            ContractType::Near => write!(f, "Near"),
+            ContractType::Dao => write!(f, "Dao"),
+            ContractType::File => write!(f, "File"),
+        }
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/contract_type.rs:34]: No need to duplicate the names of contract types as `as_str()` is already implemented.

```suggestion
        write!(f, "{}", self.as_str())
```

File: eth2near/eth2near-block-relay-rs/src/contract_type.rs [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+    }
+}
+
+impl ContractType {
+    pub fn as_str(&self) -> &str {
+        match self {
+            ContractType::Near => "Near",
+            ContractType::Dao => "Dao",
+            ContractType::File => "File",
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/contract_type.rs:43]:
```suggestion
            ContractType::Near => "NEAR",
            ContractType::Dao => "DAO",
            ContractType::File => "File",
```

File: eth2near/eth2near-block-relay-rs/src/contract_type.rs [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+impl FromStr for ContractType {
+    type Err = IncorrectContractType;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Near" => Ok(ContractType::Near),
+            "Dao" => Ok(ContractType::Dao),
+            "File" => Ok(ContractType::File),
+            _ => Err(IncorrectContractType),
+        }
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/contract_type.rs:57]: It's better to parse the lowercase version, which allows more invariants.

```suggestion
        match s.to_lowercase().as_str() {
            "near" => Ok(ContractType::Near),
            "dao" => Ok(ContractType::Dao),
            "file" => Ok(ContractType::File),
            _ => Err(IncorrectContractType),
        }
```

File: eth2near/eth2-contract-init/src/config.rs [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+
+    fn check_urls(&self) {
+        // check `beacon_endpoint`
+        Url::parse(&self.beacon_endpoint).expect("Incorrect beacon endpoint");
+
+        // check `eth1_endpoint`
+        Url::parse(&self.eth1_endpoint).expect("Incorrect eth1 endpoint");
+
+        // check `near_endpoint`
+        Url::parse(&self.near_endpoint).expect("Incorrect near endpoint");
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/config.rs:77]:
```suggestion
        // check `eth1_endpoint`
        Url::parse(&self.eth1_endpoint).expect("Incorrect ETH1 endpoint");

        // check `near_endpoint`
        Url::parse(&self.near_endpoint).expect("Incorrect NEAR endpoint");
```

File: eth2near/eth2-contract-init/src/config.rs [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    fn check_account_id(&self) {
+        let near_rpc_client = NearRPCClient::new(&self.near_endpoint);
+
+        // check `signer_account_id`
+        self.signer_account_id.parse::<near_sdk::AccountId>().expect("Incorrect signature account");
+        if !near_rpc_client
+            .check_account_exists(&self.signer_account_id)
+            .expect("Error on checking signature account existence")
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/config.rs:87]:
```suggestion
        self.signer_account_id.parse::<near_sdk::AccountId>().expect("Incorrect signer account ID");
        if !near_rpc_client
            .check_account_exists(&self.signer_account_id)
            .expect("Error on checking signer account ID existence")
```

File: eth2near/eth2-contract-init/src/config.rs [hunk 25]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            panic!("Signer account id doesn't exist on NEAR network");
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/config.rs:89]:
```suggestion
            panic!("Signer account ID doesn't exist on NEAR network");
```

File: eth2near/eth2-contract-init/src/config.rs [hunk 26]
```diff
@@ -0,0 +1,112 @@
...
+
+        // check `contract_account_id`
+        self.contract_account_id.parse::<near_sdk::AccountId>().expect("Incorrect contract account id");
+        if !near_rpc_client
+            .check_account_exists(&self.contract_account_id)
+            .expect("Error on checking contract account existence")
+        {
+            panic!("Contract account id doesn't exist on NEAR network");
+        }
+    }
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/config.rs:111]:
```suggestion
        if let Some(trusted_signature) = self.trusted_signature.clone() {
            trusted_signature.parse::<near_sdk::AccountId>().expect("Incorrect contract account ID");
            if !near_rpc_client
                .check_account_exists(&trusted_signature)
                .expect("Error on checking trusted signer account ID existence")
            {
                panic!("Trusted signer account ID doesn't exist on NEAR network");
            }
        }

        // check `contract_account_id`
        self.contract_account_id.parse::<near_sdk::AccountId>().expect("Incorrect contract account ID");
        if !near_rpc_client
            .check_account_exists(&self.contract_account_id)
            .expect("Error on checking contract account ID existence")
        {
            panic!("Contract account ID doesn't exist on NEAR network");
        }
    }
```

File: eth2near/eth2-contract-init/src/config.rs [hunk 27]
```diff
@@ -0,0 +1,112 @@
...
+
+    pub validate_updates: Option<bool>,
+
+    pub verify_bls_signature: Option<bool>,
+
+    pub hashes_gc_threshold: Option<u64>,
+
+    pub max_submitted_blocks_by_account: Option<u32>,
+
+    pub trusted_signature: Option<String>,
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/config.rs:49]: Should be `trusted_signer_account_id`

File: eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs [hunk 28]
```diff
@@ -149,7 +148,7 @@ impl Eth2NearRelay {
             eth2near_relay
                 .eth_client_contract
                 .register_submitter()
-                .unwrap();
+                .expect("Error during register submitter");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs:151]:
```suggestion
                .expect("Error on registering the submitter");
```

File: eth2near/eth2near-block-relay-rs/src/main.rs [hunk 29]
```diff
@@ -58,7 +47,7 @@ fn get_dao_contract_wrapper(config: &Config) -> Box<dyn ContractWrapper> {
         &config.near_endpoint,
         &config.signer_account_id,
         &config.path_to_signer_secret_key,
-        &dao_contract_account_id.unwrap(),
+        &dao_contract_account_id.expect("No dao contract account provided for relay running in DAO mode"),
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/main.rs:50]:
```suggestion
        &dao_contract_account_id.expect("No DAO contract account ID provided for relay running in DAO mode"),
```

File: eth2near/eth2near-block-relay-rs/src/main.rs [hunk 30]
```diff
@@ -91,32 +80,26 @@ fn init_log(args: &Arguments, config: &Config) {
...
     log::set_boxed_logger(Box::new(SimpleLogger::new(path_to_log_file)))
         .map(|()| log::set_max_level(log_level_filter))
-        .unwrap();
+        .expect("Error on logger creation");
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args = Arguments::parse();
-    let config = Config::load_from_toml(args.config.clone().try_into().unwrap());
+    let config = Config::load_from_toml(args.config.clone().try_into().expect("Error in config parsing"));
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/main.rs:93]:
```suggestion
    let config = Config::load_from_toml(args.config.clone().try_into().expect("Error on config parsing"));
```

File: eth2near/eth2near-block-relay-rs/src/prometheus_metrics.rs [hunk 31]
```diff
@@ -136,7 +136,7 @@ pub fn run_prometheus_service(port: u16) {
 
     let metrics_route = warp::path!("metrics").and_then(metrics_handler);
 
-    let rt = Runtime::new().unwrap();
+    let rt = Runtime::new().expect("Error on creation Runtime for prometheus service");
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/prometheus_metrics.rs:139]:
```suggestion
    let rt = Runtime::new().expect("Error on creating runtime for Prometheus service");
```

File: eth2near/contract_wrapper/src/lib.rs (hunk 3 above)
Comment (olga24912) [reply eth2near/contract_wrapper/src/lib.rs]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/d817c36c4d6151b2535e1d2fd541011d6e542284

File: eth2near/logger/Cargo.toml (hunk 4 above)
Comment (olga24912) [reply eth2near/logger/Cargo.toml]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/4683dcd634d532d10f8c6f5b29479f1e224be26e

File: eth2near/eth2near-block-relay-rs/Cargo.toml (hunk 5 above)
Comment (olga24912) [reply eth2near/eth2near-block-relay-rs/Cargo.toml]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/4683dcd634d532d10f8c6f5b29479f1e224be26e

File: eth2near/contract_wrapper/src/eth_network_enum.rs (hunk 7 above)
Comment (olga24912) [reply eth2near/contract_wrapper/src/eth_network_enum.rs]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/e4e2d1194f7d4e6811452999d30c08b4ec39e34e

File: eth2near/contract_wrapper/src/near_network_enum.rs (hunk 9 above)
Comment (olga24912) [reply eth2near/contract_wrapper/src/near_network_enum.rs]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/e4e2d1194f7d4e6811452999d30c08b4ec39e34e

File: eth2near/contract_wrapper/src/eth_network_enum.rs (hunk 8 above)
Comment (olga24912) [reply eth2near/contract_wrapper/src/eth_network_enum.rs]: fixed https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/22da4aab31bb1aeee58e53d98ac413506c849b5c

File: eth2near/contract_wrapper/src/near_network_enum.rs (hunk 10 above)
Comment (olga24912) [reply eth2near/contract_wrapper/src/near_network_enum.rs]: fixed https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/22da4aab31bb1aeee58e53d98ac413506c849b5c

File: eth2near/eth2near-block-relay-rs/src/contract_type.rs (hunk 22 above)
Comment (olga24912) [reply eth2near/eth2near-block-relay-rs/src/contract_type.rs]: fixed 
https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/17bcda0b6dc5e0fd8048a69831817d232213b0ad

File: eth2near/eth2near-block-relay-rs/src/contract_type.rs (hunk 20 above)
Comment (olga24912) [reply eth2near/eth2near-block-relay-rs/src/contract_type.rs]: fixed https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/17bcda0b6dc5e0fd8048a69831817d232213b0ad

File: eth2near/eth2-contract-init/src/config.rs (hunk 27 above)
Comment (olga24912) [reply eth2near/eth2-contract-init/src/config.rs]: fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/825/commits/c924a32cd6d7bf0ee1a7d54532ebc89e25ff6f48

File: eth2near/eth2near-block-relay-rs/src/test_utils.rs [hunk 32]
```diff
@@ -200,6 +206,30 @@ fn get_config(config_for_test: &ConfigForTests) -> Config {
...
+        contract_account_id: "NaN".to_string(),
+        ethereum_network: config_for_test.network_name.clone(),
+        near_network_id: NearNetwork::Testnet,
+        output_dir: None,
+        eth_requests_timeout_seconds: Some(30),
+        validate_updates: Some(true),
+        verify_bls_signature: Some(false),
+        hashes_gc_threshold: Some(51000),
+        max_submitted_blocks_by_account: Some(8000),
+        trusted_signature: Some(eth_client_contract.get_signer_account_id().to_string()),
```
Comment (sept-en) [inline eth2near/eth2near-block-relay-rs/src/test_utils.rs:228]: should be `trusted_signer_account_id`, CI fails.

File: eth2near/eth2-contract-init/src/init_contract.rs [hunk 33]
```diff
@@ -0,0 +1,214 @@
...
+
+    if let NearNetwork::Mainnet = config.near_network_id {
+        assert!(config.validate_updates.unwrap_or(true), "The updates validation can't be disabled for mainnet");
+        assert!(config.verify_bls_signature.unwrap_or(false) || config.trusted_signer_account_id.is_some(), "The client can't be executed in the trustless mode without BLS sigs verification on Mainnet");
+    }
+
+    let beacon_rpc_client = BeaconRPCClient::new(
+        &config.beacon_endpoint,
+        config.eth_requests_timeout_seconds.unwrap_or(10),
+        config.eth_requests_timeout_seconds.unwrap_or(10),
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/init_contract.rs:57]: In the next PR, these values should be replaced with some config.

File: eth2near/eth2-contract-init/src/init_contract.rs [hunk 34]
```diff
@@ -0,0 +1,214 @@
...
+        light_client_snapshot.current_sync_committee,
+        next_sync_committee,
+        config.validate_updates,
+        config.verify_bls_signature,
+        config.hashes_gc_threshold,
+        config.max_submitted_blocks_by_account,
+        trusted_signature,
+    );
+
+    thread::sleep(time::Duration::from_secs(30));
```
Comment (sept-en) [inline eth2near/eth2-contract-init/src/init_contract.rs:134]: In the next PR, these values should be replaced with some config.

**Review by sept-en — APPROVED** [review]

File: eth2near/eth2-contract-init/src/init_contract.rs (hunk 33 above)
Comment (sept-en) [reply eth2near/eth2-contract-init/src/init_contract.rs:57]: Created #840

File: eth2near/eth2-contract-init/src/init_contract.rs (hunk 34 above)
Comment (sept-en) [reply eth2near/eth2-contract-init/src/init_contract.rs:134]: Created https://github.com/aurora-is-near/rainbow-bridge/issues/840

**Pull request merged**